src/main/java/uk/gov/companieshouse/web/accounts/service/smallfull/impl/BalanceSheetServiceImpl.java

```diff
@@ -1,6 +1,24 @@
 package uk.gov.companieshouse.web.accounts.service.smallfull.impl;
 
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.web.accounts.model.smallfull.BalanceSheet;
 import uk.gov.companieshouse.web.accounts.service.smallfull.BalanceSheetService;
+import uk.gov.companieshouse.web.accounts.transformer.smallfull.BalanceSheetTransformer;
 
+@Component
 public class BalanceSheetServiceImpl implements BalanceSheetService {
+
+    private static BalanceSheetTransformer transformer = new BalanceSheetTransformer();
```

Comment:
Is there a reason we are adding the transformer as a static field? and instead not autowiring it into the class?

Reply:
We want it to be static due to the nature of the Transformer (it only needs to be initialised once as nothing differs per Service). We could qualify it with `final` too to help indicate this.

The counter-argument is that we program to an interface, use `@Autowired`, and don't make it static.

Reply:
I guess the case for the static final is a good one which definitely makes sense, however have you thought a way around this when testing? As you will not be able to mock the transformer response via Mockito.

Reply:
Good point. That's a problem which we've yet to solve, though for the purpose of this work, the static member feels like a logical approach and we'll work around mocking when we get to it.

Reply:
:+1:

Reply:
Folks - why not make the getBalanceSheet method static and the call it directly BalanceSheetTransformer.getBalanceSheet() instead of having BalanceSheetTransformer as an instance member? Would this cause any thread safety issues?

Reply:
I don't think it poses any thread safety issues as static guarantees it will be thread safe. The bigger concern from my end is more the mocking of that method as it will not be possible via Mockito.

Reply:
Agreed Chris, though we shouldn't let unit tests define our code.

Reply:
I believe you can mock static classes using PowerMockito.  I think the uk.gov.ch.cap.server.sail.SailRegisterCommandTest in the chips repo is an example that mocks the use of the SailMarshaller class. Might be worth a look.

Reply:
>though we shouldn't let unit tests define our code.
In a general sense possibly not, however, the reason why TDD is popular is because it allows you to design the code in an efficient way which also allows you to test it. In this case I feel it may show a problem with our code which is what unit tests do, especially when done in a TDD way. However, there should be nothing wrong with writing the tests, you just won't be able to mock the transformer response, instead you will have to pass in a valid object for it to transform, which could raise the point of should we be also testing the transformer in the service class. The consensus normally is no, but like I said earlier, as long as we think about these things now instead of layer, we are sure to iron out as many potential problems as possible :)

Additionally, regarding the PowerMockito usage @ljgeorge, look above for reasons for why we are trying not to go into the PowerMock direction.

Reply:
We avoid `PowerMock` where we can, but will we need to mock it? We initialise the Transformer as a static member, and it's sole purpose is to take an object of one type and turn it to another type. There's nothing intricate going on there, and nothing complicated. If we look at the code above, the `getBalanceSheet` method would, at full working order, take in 2 objects (Lets call them `Foo` and `Bar`), which it would get from the `API SDK`. The interaction with the SDK would be mocked, using something like `thenReturn` in `Mockito` we would be able to return a valid instantiated object of type `Foo` or `Bar` and use those in the static class.

Reply:
👍 That's fine.